kratos/elements/levelset_convection_element_simplex_algebraic_stabilization.h

```diff
@@ -0,0 +1,434 @@
+    {
+        KRATOS_TRY
+        return Element::Pointer(new LevelSetConvectionElementSimplexAlgebraicStabilization(NewId, GetGeometry().Create(ThisNodes), pProperties));
+        KRATOS_CATCH("");
+    }
+
+    void CalculateLocalSystem(MatrixType& rLeftHandSideMatrix, VectorType& rRightHandSideVector, const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_TRY
+
+        if (rLeftHandSideMatrix.size1() != TNumNodes)
+            rLeftHandSideMatrix.resize(TNumNodes, TNumNodes, false); //false says not to preserve existing storage!!
+
+        if (rRightHandSideVector.size() != TNumNodes)
+            rRightHandSideVector.resize(TNumNodes, false); //false says not to preserve existing storage!!
+
+        const double delta_t = rCurrentProcessInfo[DELTA_TIME];
+        const double dt_inv = 1.0 / delta_t;
+        const double theta = rCurrentProcessInfo.Has(TIME_INTEGRATION_THETA) ? rCurrentProcessInfo[TIME_INTEGRATION_THETA] : 0.5;
+
+        ConvectionDiffusionSettings::Pointer my_settings = rCurrentProcessInfo.GetValue(CONVECTION_DIFFUSION_SETTINGS);
+        const Variable<double>& rUnknownVar = my_settings->GetUnknownVariable();
+        const Variable<array_1d<double, 3 > >& rConvVar = my_settings->GetConvectionVariable();
+        const Variable<array_1d<double, 3 > >& rGradVar = my_settings->GetGradientVariable();
+
+        //getting data for the given geometry
+        BoundedMatrix<double, TNumNodes, TDim > DN_DX;
+        array_1d<double, TNumNodes > N;
+        double Volume;
+        GeometryUtils::CalculateGeometryData(GetGeometry(), DN_DX, N, Volume);
+
+        //here we get all the variables we will need
+        array_1d<double,TNumNodes> phi, phi_old;
+        array_1d< array_1d<double,3 >, TNumNodes> v, vold;
+
+        array_1d<double,3 > X_mean_tmp = ZeroVector(3);
+        array_1d<double,3 > grad_phi_mean_tmp = ZeroVector(3);
+        array_1d< array_1d<double,3 >, TNumNodes> X_node;
+        double phi_mean_old = 0.0;
+        double phi_mean = 0.0;
+
+        for (unsigned int i = 0; i < TNumNodes; i++)
+        {
+            phi[i] = GetGeometry()[i].FastGetSolutionStepValue(rUnknownVar);
+            phi_old[i] = GetGeometry()[i].FastGetSolutionStepValue(rUnknownVar,1);
+
+            v[i] = GetGeometry()[i].FastGetSolutionStepValue(rConvVar);
+            vold[i] = GetGeometry()[i].FastGetSolutionStepValue(rConvVar,1);
+
+            X_mean_tmp += GetGeometry()[i].Coordinates();
+            X_node[i] = GetGeometry()[i].Coordinates();
+
+            grad_phi_mean_tmp += GetGeometry()[i].GetValue(rGradVar);
+
+            phi_mean_old += GetGeometry()[i].FastGetSolutionStepValue(rUnknownVar,1);
+            phi_mean += GetGeometry()[i].FastGetSolutionStepValue(rUnknownVar);
+        }
+
+        phi_mean /= static_cast<double>(TNumNodes);
+        phi_mean_old /= static_cast<double>(TNumNodes);
```

Comment:
I think the cast is automatically done to a double. Did it throw a warning without this?